apps/subject/models.py

```diff
@@ -81,6 +81,16 @@ def get_semester_to_default_import(cls):
         return cls.objects.filter(courseDesciptionSubmission__lt=now) \
                           .order_by("courseDesciptionSubmission").last()
 
+    @classmethod
+    def get_offsetted_semester(cls, original_year: int, original_semester: int, offset: int):
+        # TODO: Change to get and resturn Semester after adding summer/winter semester
+        temp_semester = original_semester + offset
+        year_diff = (temp_semester - 1) // 4
+
+        new_year = original_year + year_diff
+        new_semester = temp_semester - year_diff * 4
```

Comment:
```suggestion
        year_diff, new_semester = divmod(original_semester + offset, 4)
        new_year = original_year + year_diff
```
요렇게는 어떨까요? 계산이 맞을지는 모르겠는데 대충 이런 느낌?

Reply:
semester가 0~3이면 맞는데 현재 1~4로 되어 있어서 계산이 맞지 않는거같아요
예를들어 2021,3에 offset 1을 적용하면 2021,4가 되어야 하는데 2022,0이 됩니다
divmod를 쓰고자 한다면 밑에
```python
if next_semester == 0:
    next_year -= 1
    next_semester = 4
```
를 붙이면 되겠지만 원본 코드보다 더 깔끔한지 의문이네요
혹시 더 좋은 코드 있으면 제안 부탁드립니다

Reply:
그렇다면 조금 길긴 한데 이게 직관적이긴 한 거 같아요 로직 이해하기에
```python
def next_semester(year, semester):
  if semester === 4:
    return year + 1, 1
  else:
    return year, semester + 1

def prev_semester(year, semester):
  if semester === 1:
    return year - 1, 4
  else:
    return year, semester - 1

def add_semester(year, semester, delta):
  op = next_semester if delta >= 0 else prev_semester
  for i in range(abs(delta)):
    year, semester = op(year, semester)
  return year, semester
```

Reply:
괜찮네요. prev_semeste랑 next_semester도 자체적으로 쓸만할 것 같습니다. 코드만 살짝 수정해서 반영했습니다.
a3433598